makefile

```diff
@@ -104,8 +104,11 @@ endif
 ifdef STAN_OPENCL
 STAN_FLAG_OPENCL=_opencl
 endif
+ifdef STAN_NO_RANGE_CHECKS
+STAN_FLAG_NO_RANGE_CHECKS=_nochecks
```

Comment:
isn't it  better to define an else branch here which sets the name to a defined name when we do not have range checks off?

Reply:
Yeah, I guess that would be a bit safer. Whenever we dont use it, I think the best way is to just have no name so that whenever none of these flags are used, it will make a main.o file, without the suffixes.

Reply:
Fixed.